tidy3d/components/simulation.py

```diff
@@ -242,6 +242,58 @@ def bloch_with_symmetry(cls, val, values):
                 )
         return val
 
+    # pylint: disable=too-many-locals
+    @pydantic.validator("boundary_spec", always=True)
+    def plane_wave_boundaries(cls, val, values):
+        """Error if there are plane wave sources incompatible with boundary conditions."""
+        boundaries = val.to_list
+        sources = values.get("sources")
+        size = values.get("size")
+        sim_medium = values.get("medium")
+        structures = values.get("structures")
+        for src_idx, source in enumerate(sources):
+            if not isinstance(source, PlaneWave):
+                continue
+
+            _, tan_dirs = cls.pop_axis([0, 1, 2], axis=source.injection_axis)
+            medium_set = cls.intersecting_media(source, structures)
+            medium = medium_set.pop() if medium_set else sim_medium
+
+            for tan_dir in tan_dirs:
+                boundary = boundaries[tan_dir]
+
+                # check the PML/absorber + angled plane wave case
+                num_pml = sum(isinstance(bnd, (PML, StablePML, Absorber)) for bnd in boundary)
+                if num_pml > 0 and source.angle_theta != 0:
+                    raise SetupError(
+                        "Angled plane wave sources are not compatible with the PML boundary along "
+                        f"dimension {tan_dir}. Either set the source ``angle_theta`` to ``0``, "
+                        "or use Bloch boundaries that match the source angle."
+                    )
+
+                # check the Bloch boundary + angled plane wave case
+                num_bloch = sum(isinstance(bnd, BlochBoundary) for bnd in boundary)
+                if num_bloch > 0:
+                    # make a dummy Bloch boundary to check for correctness
+                    dummy_bnd = BlochBoundary.from_source(
+                        source=source, domain_size=size[tan_dir], axis=tan_dir, medium=medium
+                    )
+                    expected_bloch_vec = dummy_bnd.bloch_vec
+                    if boundary[0].bloch_vec != expected_bloch_vec:
+                        test_val = np.real(expected_bloch_vec - boundary[0].bloch_vec)
+                        if test_val.is_integer() and test_val != 0:
+                            log.warning(
+                                f"The Bloch vector along dimension {tan_dir} is not in the first "
+                                "Brillouin zone."
```

Comment:
It seems that one of my comments somehow got lost (unfortunately probably the longest one for me to type...)

To clarify, when when I mentioned the first Brillouin zone I meant something else: we *do* allow the Bloch vector to be outside of it, and we don't need to warn about this. This just means that `boundary.bloch_vec` is not restricted to be between -1 and 1. Here what you are checking instead is whether the source in-plane k-vector matches the boundary bloch vector exactly, or only modulo 1. The warning is a bit hard to phrase, but something along the lines of:

```
The wave vector of source index {src_idx} along dimension {tan_dir} is equal to the Bloch vector of the simulation boundaries along that dimension plus an integer reciprocal lattice vector. If using a 'DiffractionMonitor', diffraction order 0 will not correspond to the angle of propagation of the source.
```

Reply:
Oh I see. So for my understanding: the old wording makes it sound as though we are prodding the user _not_ to set Bloch vectors outside the first Brillouin zone - but in fact there's no issue with that. The warning is instead meant to remind the user of the implications when interpreting  the results, in particular the diffraction orders - correct?

Reply:
Yeah, exactly